drivers/char/tpm/tpm-interface.c

```diff
@@ -953,6 +953,11 @@ int tpm_do_selftest(struct tpm_chip *chip)
 	loops = jiffies_to_msecs(duration) / delay_msec;
 
 	rc = tpm_continue_selftest(chip);
+	if (rc == TPM_ERR_INVALID_POSTINIT) {
```

Comment:
You should probably log this condition with something like
`dev_info(&chip->dev, "TPM not ready (%d)\n", rc);`